ros/src/computing/perception/detection/lidar_tracker/packages/lidar_imm_ukf_pda_track/nodes/imm_ukf_pda/imm_ukf_pda.cpp

```diff
@@ -19,120 +47,157 @@ ImmUkfPda::ImmUkfPda()
   private_nh_.param<double>("gate_probability", gate_probability_, 0.99);
   private_nh_.param<double>("detection_probability", detection_probability_, 0.9);
   private_nh_.param<double>("distance_thres", distance_thres_, 99);
-  private_nh_.param<double>("static_distance_thres", static_distance_thres_, 3.0);
+  private_nh_.param<double>("static_velocity_thres", static_velocity_thres_, 0.5);
+  private_nh_.param<double>("prevent_explosion_thres", prevent_explosion_thres_, 1000);
+  private_nh_.param<bool>("use_sukf", use_sukf_, false);
+  private_nh_.param<bool>("is_debug", is_debug_, false);
+}
 
-  init_ = false;
+void ImmUkfPda::run()
+{
+  pub_jskbbox_array_ =
+      node_handle_.advertise<jsk_recognition_msgs::BoundingBoxArray>("/detection/lidar_tracker/bounding_boxes", 1);
+  pub_object_array_ = node_handle_.advertise<autoware_msgs::DetectedObjectArray>("/detection/lidar_tracker/objects", 1);
 
-  tf::TransformListener* lr(new tf::TransformListener);
-  tf_listener_ = lr;
+  // for debug
+  pub_points_array_ =
+      node_handle_.advertise<visualization_msgs::MarkerArray>("/detection/lidar_tracker/debug_points_markers", 1);
+  pub_texts_array_ =
+      node_handle_.advertise<visualization_msgs::MarkerArray>("/detection/lidar_tracker/debug_texts_markers", 1);
 
-  sub_cloud_array_ = node_handle_.subscribe("cloud_clusters", 1, &ImmUkfPda::callback, this);
-  pub_jskbbox_array_ = node_handle_.advertise<jsk_recognition_msgs::BoundingBoxArray>("/bounding_boxes_tracked", 1);
-  pub_object_array_ = node_handle_.advertise<autoware_msgs::DetectedObjectArray>("/detected_objects", 1);
+  sub_detected_array_ = node_handle_.subscribe("/detection/lidar_detector/objects", 1, &ImmUkfPda::callback, this);
 }
 
-void ImmUkfPda::callback(const autoware_msgs::CloudClusterArray& input)
+void ImmUkfPda::callback(const autoware_msgs::DetectedObjectArray& input)
 {
-  autoware_msgs::CloudClusterArray transformed_input;
+  autoware_msgs::DetectedObjectArray transformed_input;
   jsk_recognition_msgs::BoundingBoxArray jskbboxes_output;
   autoware_msgs::DetectedObjectArray detected_objects_output;
 
   // only transform pose(clusteArray.clusters.bouding_box.pose)
   transformPoseToGlobal(input, transformed_input);
   tracker(transformed_input, jskbboxes_output, detected_objects_output);
   transformPoseToLocal(jskbboxes_output, detected_objects_output);
-
   pub_jskbbox_array_.publish(jskbboxes_output);
   pub_object_array_.publish(detected_objects_output);
 }
 
-void ImmUkfPda::transformPoseToGlobal(const autoware_msgs::CloudClusterArray& input,
-                                            autoware_msgs::CloudClusterArray& transformed_input)
+void ImmUkfPda::relayJskbbox(const autoware_msgs::DetectedObjectArray& input,
+                             jsk_recognition_msgs::BoundingBoxArray& jskbboxes_output)
 {
+  jskbboxes_output.header = input.header;
+  for (size_t i = 0; i < input.objects.size(); i++)
```

Comment:
range-based for is recommended where index "i" is used only for iterating one array.